File: C++/WarmUpChallenges/Repeated Strings/strings.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the repeatedString function below.
long repeatedString(string s, long n) {
    //ll n;
    long long count=0;
    cin >> n;
    for(long long i=0;i<s.size();i++){
        if(s[i] == 'a')
            count++;
    }
    long long ans =0 ;
    ans = n/s.size();
    cout << ans << " " << count << endl;
    ans*=count;
    cout << n%s.size() << endl;
    if((n%s.size())){
        long long r = n%s.size();
        for(long long i=0;i<r;i++){
            if(s[i] == 'a') ans++;
        }
    }

    return ans;

}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    long n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    long result = repeatedString(s, n);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: Python/Repeated Strings/strings.py
#!/bin/python

import math
import os
import random
import re
import sys

# Complete the repeatedString function below.
def repeatedString(s, n):
    count = 0
    ans = 0
    c=0
    for i in s:
        if i == 'a':
            count+=1
    ans = (n/len(s))*count
    r = n%len(s)
    #print count
    print ans
    print r
    if(n%len(s)):
        for i in s:
            c+=1
            if i == 'a' and c<=r:
                ans+=1
    return ans

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = raw_input()

    n = int(raw_input())

    result = repeatedString(s, n)

    fptr.write(str(result) + '\n')

    fptr.close()
